### PR TITLE
feat: FO-70 | Workflow Python ds-pack images

### PR DESCRIPTION
**Pull request by saurabh0402**
## Description
This PR adds a new `workflow-python-ds-pack-v1` image. The image is an extension of the `workflow-python-barebones-v1` image with the following packages installed on top of it
```
snowflake
snowflake-connector-python[pandas]
sklearn
pandas
```

## Review Checks
- [x] 📝 The commit message is clear and descriptive
- [x] 🔐 The Security Considerations section in the PR description is complete
- [ ] ✅ Tests for the changes have been added and run successfully including the new changes
- [x] 📄 Documentation has been added / updated (for bug fixes / features)

## Dependencies
None

**Does this PR introduce a breaking change?** 
- [ ] Yes
- [x] No

## Security Considerations
- We have added the required non-root user to the image.
- No secret or peak-specific data is added to the image.

## Types of change
- [ ] 🐛 Bugfix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 Adding or updating configuration files, development scripts etc.
- [ ] ♻️ Refactoring (no functional changes, no API changes)
- [ ] 🧹 Chore (removing redundant files, fixing typos, etc.)
- [ ] 📄 Documentation Update 
- [ ] ❓ Other (if none of the other choices apply)

## Testing
- Check that we are able to build the image and use it in workflows.

### CODE REVIEW

**Comment by vanessavirgo94** [comment]
Can we pause this merge? Jason Gofford has some ideas that could help with this, but waiting on him to finish his merge

File: workflow/python/barebones/v1/README.md [hunk 1]
```diff
@@ -29,12 +29,14 @@ sqlalchemy     1.4.36
...
 ```
-docker build . -t workflow-python-barebones-v1 -build-arg PEAK_USER_ID=8877
+docker build . -t workflow-python-barebones-v1 --build-arg PEAK_USER_ID=8877
 ```
 
 ## Using the Image
 - The image can be directly used by using it in the workflow step form.
-- If you need to install a few more dependencies, or add some use case-specific environment variables into the image, the image can easily to extended.
\ No newline at end of file
+- If you need to install a few more dependencies, or add some use case-specific environment variables into the image, the image can easily to extended.
```
Comment (peak-jiteshahuja) [inline workflow/python/barebones/v1/README.md:42]:
```suggestion
- If you need to install a few more dependencies, or add some use case-specific environment variables into the image, the image can easily be extended.
```

File: workflow/python/ds-pack/v1/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+FROM python:3.8.13
+
+RUN apt-get update
```
Comment (peak-jiteshahuja) [inline workflow/python/ds-pack/v1/Dockerfile:3]: Hey @saurabh0402 . Can we pl combine lines 3 and 4 to reduce docker layers ?

File: workflow/r/barebones/v1/README.md [hunk 3]
```diff
@@ -32,16 +32,22 @@ odbc                    1.3.3
 ### ODBC Setup
 - The `odbcinst.ini` file exists in the `/home/peak-user` directory, and the `ODBCSYSINI` env has been set accordingly.
 
+You can find more details about setting up ODBC [here](https://docs.aws.amazon.com/redshift/latest/mgmt/configure-odbc-connection.html).
```
Comment (peak-jiteshahuja) [inline workflow/r/barebones/v1/README.md:35]: Hey aren't these only for redshift ? I guess we also need to add links for snowflake here ?

**Comment by peak-jiteshahuja** [comment]
> Can we pause this merge? Jason Gofford has some ideas that could help with this, but waiting on him to finish his merge

Hey @vanessavirgo94 - Any update on this ?

**Comment by vanessavirgo94** [comment]
@peak-jiteshahuja Not as I know. Think the code I was referring to might take longer than this, so we could leave this for now and come back to it?

**Comment by peak-jiteshahuja** [comment]
> Gofford has some ideas that could help with this,

Sure. Thanks

**Comment by peak-jiteshahuja** [comment]
> @peak-jiteshahuja Not as I know. Think the code I was referring to might take longer than this, so we could leave this for now and come back to it?
Sure. Thanks

**Comment by peak-jiteshahuja** [comment]
Have added some comments which can be addressed in the next PR. Approving this PR.

**Review by peak-jiteshahuja — APPROVED** [review]

File: workflow/python/ds-pack/v1/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+boto3==1.23.7
+pandas==1.4.2
+psycopg2==2.9.3
+snowflake==0.0.3
+snowflake-connector-python[pandas]==2.7.7
+sklearn==0.0
```
Comment (vanessavirgo94) [inline workflow/python/ds-pack/v1/requirements.txt:6]: Should this be version '0.0'? On a workspace I have I used '1.0.2'?

File: workflow/python/ds-pack/v1/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+boto3==1.23.7
+pandas==1.4.2
+psycopg2==2.9.3
+snowflake==0.0.3
+snowflake-connector-python[pandas]==2.7.7
```
[the remark this answers is not in the document]
Comment (vanessavirgo94) [reply workflow/python/ds-pack/v1/requirements.txt:5]: This is what I have in a workspace FYI

snowflake==0.0.3
snowflake-connector-python==2.7.6

**Comment by vanessavirgo94** [comment]
As a question - why isn't there snowflake connectors in `barebones/v1` image?

**Comment by vanessavirgo94** [comment]
FYI this is what Jason is working on https://github.com/PeakBI/ds-sqltools/pull/6 - could be useful!

File: workflow/python/ds-pack/v1/requirements.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (peak-jiteshahuja) [reply workflow/python/ds-pack/v1/requirements.txt:5]: Hey @vanessavirgo94 - Seems like the newest version for `snowflake-connector-python[pandas]` is 2.7.8. Is it okay if we go ahead with the new version or do we want to continue using `2.7.6` instead ?

File: workflow/python/ds-pack/v1/requirements.txt (hunk 4 above)
Comment (peak-jiteshahuja) [reply workflow/python/ds-pack/v1/requirements.txt:6]: Hey @vanessavirgo94. Is it okay if replace sklearn  with scikit-learn (1.1.1 - latest one) as sklearn pypi itself recommends using scikit-learn  (Ref- https://pypi.org/project/sklearn/)

**Comment by peak-jiteshahuja** [comment]
> FYI this is what Jason is working on https://github.com/PeakBI/ds-sqltools/pull/6 - could be useful!

Hey @vanessavirgo94 - Thanks for the update. Sadly i am unable to access it atm. I'l surely go through the changes once. Would it be okay if we can address those in the next PR unless they are very important. In that case i am happy to have them addressed in this PR.

**Review by parul157 — APPROVED** [review]

File: workflow/python/ds-pack/v1/requirements.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (vanessavirgo94) [reply workflow/python/ds-pack/v1/requirements.txt:5]: Yeah it should be fine

File: workflow/python/ds-pack/v1/requirements.txt (hunk 4 above)
Comment (vanessavirgo94) [reply workflow/python/ds-pack/v1/requirements.txt:6]: Yeah that is fine with me

**Review by vanessavirgo94 — APPROVED** [review]

**Pull request merged**